daemon/prune.go

```diff
@@ -152,7 +152,9 @@ func (daemon *Daemon) VolumesPrune(ctx context.Context, pruneFilters filters.Arg
 		return nil
 	}
 
-	err = daemon.traverseLocalVolumes(pruneVols)
+	if err := daemon.traverseLocalVolumes(pruneVols); err == context.Canceled {
+		return rep, nil
+	}
 
 	return rep, err
```

Comment:
this is now returning a different error than before, because the `err = daemon.traverseLocalVolumes()` is now moved inside the `if ...`